java/vector/src/main/java/org/apache/arrow/vector/GenerateSampleData.java

```diff
@@ -0,0 +1,337 @@
+/*******************************************************************************
+
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ ******************************************************************************/
+
+package org.apache.arrow.vector;
+
+import java.math.BigDecimal;
+import java.nio.charset.Charset;
+
+/**
+ * Helper class to generate test data for Nullable fixed and variable
+ * width scalar vectors. Previous implementations of java vector classes
+ * provided generateTestData(now deprecated) API to populate the vector
+ * with sample data. This class should be used for that purpose.
+ */
+public class GenerateSampleData {
```

Comment:
Should this class be under test or tools?

Reply:
Ideally under tools but for some weird reason, I wasn't able to resolve the dependency w.r.t arrow tools in Dremio -- I was definitely missing something but for saving time just put it here.

Reply:
I prefer if we can move this out of vector public class. If we don't have time to do it here, can we track this somewhere so we don't forget to move it before release?

Reply:
Ping @siddharthteotia. Looks like this class is not used anywhere. Do we need this?

Reply:
We use this in Dremio

Reply:
If it's only used for Dremio testing, does it make more sense to stay in Dremio codebase?

Reply:
This is not something new. Vector classes have had the API generateTestData() from the beginning. As part of refactoring, we decided to move it out of the main API of vectors and provide a separate API for populating vectors with dummy data.

Reply:
The old generateTestData() API in vectors is deprecated and the class description implies that any users of the former should now be using GenerateSampleData.

Reply:
I see. It makes more sense now, thanks for the explanation. Let's keep it then.